sycl/source/detail/config.hpp

```diff
@@ -614,6 +614,16 @@ template <> class SYCLConfig<SYCL_ENABLE_FUSION_CACHING> {
   }
 };
 
+template <> class SYCLConfig<SYCL_CACHE_DISABLE_IN_MEM> {
+  using BaseT = SYCLConfigBase<SYCL_CACHE_DISABLE_IN_MEM>;
+
+public:
+  static bool get() {
+    static bool Val = BaseT::getRawValue();
```

Comment:
I was following the behavior of `SYCL_DISABLE_PARALLEL_FOR_RANGE_ROUNDING` which just disables behavior when any value is set, though I think actually it would probably be better to align this with behavior of `SYCL_CACHE_PERSISTENT`.